### PR TITLE
Add support for aliases

### LINKED ISSUE #143: Add basic support for command aliases
This is different from #92. 

This issue just tracks "we have some support for built-in aliases". Doesn't let the user change theses yet. But it does let us get closer to some PT run parity. 

* [ ] The `ListItemTemplate` needs to be able to render these aliases as a part of it
* [ ] The `ListItemViewModel` needs to have a (very likely blank) string for `Alias` 
* [ ] We need to special-case some of the builtin, top-level commands as having aliases
  * [ ] `=` -> Calculator
  * [ ] `:` -> Registry
  * [ ] `$` -> Settings
  * [ ] you
  * [ ] get
  * [ ] it
* [ ] When the user types that alias (followed by whitespace? @plante-msft for whatever the user research said), we immediately jump into that command's page

We can build on top of that for customization later

### PR DESCRIPTION
**Pull request by zadjii-msft**
Honestly, code is pretty simple.

* Add an `AliasManager` to track the alias strings bound to which commands. 
  * Only a couple aliases are built-in currently. The ones from PT Run. 
* When the text is updated on the MainPage, if it matches an alias, we'll immediately just _do that command_. 
* When we do fire off an alias, we need to clear the search text on the main page, because we _don't_ want to come back with that text populated (as opposed to something like `masto↲` which we would want to leave in the search box

Closes #143

### CODE REVIEW

File: src/modules/cmdpal/Microsoft.CmdPal.UI/Controls/SearchBar.xaml.cs [hunk 1]
```diff
@@ -112,6 +123,23 @@ private void FilterBox_KeyDown(object sender, KeyRoutedEventArgs e)
...
+        // (otherwise aliases just stop working)
+        if (FilterBox.Text.Length == 1)
+        {
+            if (CurrentPageViewModel != null)
+            {
+                CurrentPageViewModel.Filter = FilterBox.Text;
+            }
+
+            return;
+        }
```
Comment (michael-hawker) [inline src/modules/cmdpal/Microsoft.CmdPal.UI/Controls/SearchBar.xaml.cs:141]: Would you mind at least trying to pull from Main Latest locally? Just to at least confirm that the behavior is fixed?

```
https://pkgs.dev.azure.com/dotnet/CommunityToolkit/_packaging/CommunityToolkit-MainLatest/nuget/v3/index.json
```

If not, then I could at lest ensure we fix it in the Toolkit again if there is still an issue before we ship the next update to NuGet.

File: src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/AliasManager.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        if (_aliases.TryGetValue(searchText, out var alias))
+        {
+            try
+            {
+                var tlc = _tlcManager.LookupCommand(alias.CommandId);
+                if (tlc != null)
+                {
+                    WeakReferenceMessenger.Default.Send<ClearSearchMessage>();
+                    WeakReferenceMessenger.Default.Send<PerformCommandMessage>(new(new(tlc.Command)));
```
Comment (michael-hawker) [inline src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/AliasManager.cs:34]: Should these be linked somehow? I believe we have the search box tied to the page that's being displayed as navigation occurs. So if the command navigates, then that should be handled. Are there just other scenarios where when we're processing the PerformCommandMessage that the searchbox should be reacting to?

File: src/modules/cmdpal/Microsoft.CmdPal.UI/Controls/SearchBar.xaml.cs (hunk 1 above)
Comment (zadjii-msft) [reply src/modules/cmdpal/Microsoft.CmdPal.UI/Controls/SearchBar.xaml.cs:141]: @ me go get rid of the other HACK from #229 too

File: src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/AliasManager.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        return false;
+    }
+
+    private void PopulateDefaultAliases()
+    {
+        this.AddAlias(new CommandAlias(":", "com.microsoft.cmdpal.registry", true));
+        this.AddAlias(new CommandAlias("$", "com.microsoft.cmdpal.windowsSettings", true));
+        this.AddAlias(new CommandAlias("=", "com.microsoft.cmdpal.calculator", true));
+        this.AddAlias(new CommandAlias(">", "com.microsoft.cmdpal.shell", true));
+        this.AddAlias(new CommandAlias("<", "com.microsoft.cmdpal.windowwalker", true));
```
Comment (joadoumie) [inline src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/AliasManager.cs:52]: General thoughts I've been having regarding aliases that I'm not sure how to reason about just yet. This very well may be NOT the correct thread: 

1. Right now extensions have to input "Id" into their command for us to be able to identify that command - seems simple enough, except what happens if there is an Id collision? Example -- two extensions choose the exact same Id `ohibetthisisaquniqueid45663636363`. That is a really bad example, but maybe `id1`. 

2. Because of the req. that extensions manually bake in their own Id, only extensions that follow this convention would allow their commands to have aliases. As we think about the general settings situation -- this is going to get complicated because it's not just aliases anymore but also direct activation and keyboard shortcuts (in the long run). That means if an extension doesn't create an Id for a command that the user won't be able to benefit from things like creating aliases, but it feels like it should just work for all commands as a user.

3. It would also be great to be able to have aliases for applications themselves. We have that in a few mocks... maybe those would already have some unique ids though we could leverage.

**Review by joadoumie — APPROVED** [review]

File: src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/AliasManager.cs (hunk 3 above)
Comment (zadjii-msft) [reply src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/AliasManager.cs:52]: Don't overindex on the IDs. Terminal has a similar concept with the IDs for our actions, and if one isn't provided, we can always generate a secret one behind the scenes. We could use the PFN of the app the command came from, hash the title/subtitle/name, slap those together and presto - unique strings for almost anything. 

Self-assigning id's might make sense for an app developer who wants to actually manually invoke the palette (e.g. via a URI handler). Or, the spec also calls out affordances for apps to more efficiently invoke commands when we add support for stub commands, if they give us an ID. 

Picking relatively unique "namespace" strings for things like this is usually not very much of a problem. Chances are if you think your namespace will conflict with someone, it probably will, so pick something more unique. We can also manually reject IDs with the prefix `com.microsoft.cmdpal`

basically there's a lot more we can do here. This is just the beginning.

Comment (zadjii) [reply src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/AliasManager.cs:52]: That's #92 😉  I really gotta stand up an actual settings model so that we can build a settings viewmodel and settings page.

File: src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/AliasManager.cs (hunk 2 above)
Comment (zadjii-msft) [reply src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/AliasManager.cs:34]: Yea this felt odd as I was writing it. 

Aliases are a uniquely main page thing, and IMO it makes sense to clear out the main page's search text when going to an aliased command. That way, when you go back, you don't have `>` just sitting there waiting for you

For pretty much any other scenario though, I feel like you want to leave the previous page's search state intact, so that when you go back, it's still there. Like if I'm in the Obsidian notes list, and I filter to "Extensions", preview a note, then go back, I should still get "Extensions" in the search. 

So, it's only this weird wacky main page that acts like this.

File: src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/AliasManager.cs (hunk 3 above)
Comment (zadjii-msft) [reply src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/AliasManager.cs:52]: oh man I really need to force myself to open Civilization and actually start my vacation before I just go do the whole settings model

File: src/modules/cmdpal/Microsoft.CmdPal.UI/Controls/SearchBar.xaml.cs (hunk 1 above)
Comment (zadjii-msft) [reply src/modules/cmdpal/Microsoft.CmdPal.UI/Controls/SearchBar.xaml.cs:141]: It looks like with `CommunityToolkit.WinUI.Extensions" Version="8.2.241204-build.1297`, I could delete both these HACK sections out of this file. So that's good! I'll file a TODO to update to the latest toolkit version when that's available and get rid of these
